lib/at_cmd/at_cmd.c

```diff
@@ -477,9 +480,9 @@ static int at_cmd_driver_init(const struct device *dev)
 				     THREAD_PRIORITY, 0, K_NO_WAIT);
 	k_thread_name_set(socket_tid, "at_cmd_socket_thread");
 
-	initialized = true;
 	LOG_DBG("Common AT socket processing thread created");
-
+	initialized = true;
+	k_sem_give(&at_cmd_init_sem);
```

Comment:
I suggest releasing (or unlocking) before each of the two other return statements in this function. That avoids a lockup after error/early returns and a new (or simultaneous) call.

Reply:
That's right @nordic-hani; need to unlock when we `return` elsewhere.